src/runtime/vm/executable.cc

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file tvm/runtime/vm/executable.cc
+ * \brief The implementation of a virtual machine executable APIs.
+ */
+
+#include <dmlc/memory_io.h>
+#include <tvm/runtime/vm.h>
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include "serializer.h"
+
+namespace tvm {
+namespace runtime {
+namespace vm {
+
+PackedFunc Executable::GetFunction(const std::string& name,
+    const std::shared_ptr<ModuleNode>& sptr_to_self) {
+  if (name == "set_context") {
```

Comment:
I think we shouldn't add this API to executable. The contexts should be set by VMCompiler. The executable should be readonly to frontent

Reply:
yes, I thought about this as well by passing context to VM compiler. But it looks that context is more runtime related.

Reply:
moved back to vm.